### PR TITLE
[NativeAOT-LLVM] Implement more opcodes

### PR DESCRIPTION
**Pull request by SingleAccretion**
Please refer to the commit list for the exact set.

Also cleaning up typed pointer leftovers.

### CODE REVIEW

**Comment by SingleAccretion** [comment]
@dotnet/nativeaot-llvm

File: src/coreclr/jit/llvmcodegen.cpp [hunk 1]
```diff
@@ -2352,6 +2424,21 @@ void Llvm::buildCkFinite(GenTreeUnOp* ckNode)
...
+    // KEEPALIVE is used to represent implicit uses of GC-visible values, e. g.:
+    //
+    //  ObjWithFinalizer obj = new ObjWithFinalizer();
+    //  NativeResource handle = obj.NativeResource;
+    //  <-- Here the compiler could think liveness of "obj" ends and permit its finalization. -->
+    //  NativeCall(handle);
+    //  <-- We insert KeepAlive s.t. we don't finalize away "handle" while it is still in use by the native call. -->
+    //  GC.KeepAlive(obj)
+    //
+    // In the shadow stack model, this would have to be done before codegen, by spilling "obj", so we don't need to do
```
Comment (yowl) [inline src/coreclr/jit/llvmcodegen.cpp]: Is it worth putting a TODO in llvmcodegen somewhere about this?

**Review by yowl — COMMENTED** [review]
Looks good, just the one comment on `KEEPALIVE`

**Comment by yowl** [comment]
Thanks for the merge comments!

File: src/coreclr/jit/llvmcodegen.cpp (hunk 1 above)
Comment (SingleAccretion) [reply src/coreclr/jit/llvmcodegen.cpp]: I suppose it is already handled in lowering save for the SDSU case we discussed some time back. I will reword this a little.

**Review by yowl — APPROVED** [review]
Thank you.

**Comment by jkotas** [comment]
Thanks!

**Pull request merged**